src/teleport.js

```diff
@@ -0,0 +1,37 @@
+import { checkForAlpine } from './utils'
+
+const AlpineTeleportMagicMethod = {
+    start() {
+        checkForAlpine()
+        Alpine.addMagicProperty('teleport', ($el) => {
+            return function (target = document.body, $element = $el, prepend = false) {
+                // find the portal
+                const destination = document.querySelector(`[x-portal="${target}"]`)
+
+                // throw error if portal is undefined
+                if (destination === undefined) {
+                    throw Error('Undefined portal: ', destination)
+                }
+
+                // teleport right after alpine done its thing
+                setTimeout(() => {
+                    // check if prepend
+                    if (prepend) {
+                        destination.prepend($element)
+                        return
+                    }
+                    // append element to portal destination
+                    destination.append($element)
+                }, 100)
```

Comment:
👀  this `100` makes my race-condition worry-meter go off, could we poll until `__x` is defined instead?

Reply:
Yeah, i am aimed to find out how to use it as 
```js
$el.__x.$nextTickStack.push(() => destination.append($element))
```
that's where i am stuck at the moment. Any suggestion will be helpful. $el.__x is undefined in helpers

Reply:
If you do `x-init="$teleport(.. )"` Alpine will still be building the component so __x won't be available and your directive won't be evaluated yet.
I think you won't need a timeout if you do `x-init="() => $teleport(...)"`. Maybe we can just make clear in the docs that it needs to run after alpine is ready.

Reply:
Yeah, that works, but we still need a way to access `$el.__x` for the fully support reactivity. I believe if i can inject the append in `$nextTickStack`, communication between old and new place will be work both ways.